src/router/handlers/fedimint/wallet/await_deposit.rs

```diff
@@ -1,84 +1,63 @@
-}
+    use anyhow::anyhow;
+    use axum::extract::State;
+    use axum::http::StatusCode;
+    use axum::Json;
+    use axum::extract::ws::{Message, Sender};
+    use fedimint_client::ClientArc;
+    use fedimint_core::config::FederationId;
+    use fedimint_core::core::OperationId;
+    use fedimint_wallet_client::{DepositState, WalletClientModule};
+    use futures_util::{SinkExt, StreamExt};
+    use serde::{Deserialize, Serialize};
+    use serde_json::{json, Value};
+
+    use crate::error::AppError;
+    use crate::state::AppState;
+
+    #[derive(Debug, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct AwaitDepositRequest {
+        pub operation_id: OperationId,
+        pub federation_id: Option<FederationId>,
+    }
 
-async fn _await_deposit(
-    client: ClientArc,
-    req: AwaitDepositRequest,
-) -> Result<AwaitDepositResponse, AppError> {
-    let mut updates = client
-        .get_first_module::<WalletClientModule>()
-        .subscribe_deposit_updates(req.operation_id)
-        .await?
-        .into_stream();
+    #[derive(Debug, Serialize)] 
+    #[serde(rename_all = "camelCase")]
+    pub struct AwaitDepositResponse {
+        pub status: DepositState,
+    }
 
-    while let Some(update) = updates.next().await {
-        match update {
-            DepositState::Confirmed(tx) => {
-                return Ok(AwaitDepositResponse {
-                    status: DepositState::Confirmed(tx),
-                })
-            }
-            DepositState::Claimed(tx) => {
-                return Ok(AwaitDepositResponse {
-                    status: DepositState::Claimed(tx),
-                })
-            }
-            DepositState::Failed(reason) => {
-                return Err(AppError::new(
-                    StatusCode::INTERNAL_SERVER_ERROR,
-                    anyhow!(reason),
-                ))
-            }
-            _ => {}
+    pub async fn handle_ws(
+        state: AppState,
+        params: Value,
+        ws_sender: Sender,
+    ) -> Result<(), AppError> {
+        // Deserialize requested parameters
+        let req: AwaitDepositRequest = serde_json::from_value(params)
+            .map_err(|e| AppError::new(StatusCode::BAD_REQUEST, anyhow!("Invalid request: {}", e)))?;
+    
+        let client = state.get_client(req.federation_id).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+    
+        let mut updates = client
+            .get_first_module::<WalletClientModule>()
+            .subscribe_deposit_updates(req.operation_id)
+            .await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?
+            .into_stream();
+    
+        while let Some(update) = updates.next().await {
+            let message = serde_json::to_string(&update).map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
+            ws_sender.send(Message::Text(message)).await.map_err(|_| AppError::from(StatusCode::INTERNAL_SERVER_ERROR))?;
```

Comment:
again, enums and json structs > text messages. See how I structured the json responses for websockets previosuly, think it's in `ws.rs` under build response